### DISCUSSION TITLE
Best way to convert a mol from SMARTS into a normal mol?

### DISCUSSION THREAD

**Discussion by lilyminium** [rdkit/rdkit#5151] [Q&A]
What would be the best way convert a molecule created from SMARTS (with QueryAtoms) to a normal molecule with normal Atoms, retaining custom properties etc. added on? My only idea currently is manually copying `Chem.Atom(QueryAtom)` to a `RWMol` and accompanying bonds -- if anyone knows of a better way, would love to hear it :)

**Comment by wopozka** [comment]
Well, MolToSmiles would create molecule but you need to remember, that when every SMILES is actually corrects SMARTS, it is not other way round. So query atoms in SMARTS might mean many different things, and you really need to create sort of conversion table to recognize what is going on there and to create proper SMILES.
